esp/smc/SMCLib/TpWrapper.cpp

```diff
@@ -1824,31 +1824,71 @@ void CTpWrapper::getAttPath(const char* Path,StringBuffer& returnStr)
 
 void CTpWrapper::getServices(double version, const char* serviceType, const char* serviceName, IArrayOf<IConstHPCCService>& services)
 {
-    Owned<IPropertyTreeIterator> itr = getGlobalConfigSP()->getElements("services");
-    ForEach(*itr)
+    Owned<IEnvironmentFactory> factory = getEnvironmentFactory(false);
+    Owned<IConstEnvironment> env = factory->openEnvironment();
+    Owned<IPropertyTree> envRoot = &env->getPTree();
+
+    if (isEmptyString(serviceType) || strieq(serviceType, "roxie"))
+        getRoxieServices(envRoot, serviceName, services);
+    getESPServices(envRoot, serviceType, serviceName, services);
+}
+
+void CTpWrapper::getRoxieServices(IPropertyTree* environmentRoot, const char* serviceName, IArrayOf<IConstHPCCService>& services)
+{
+    if (!isEmptyString(serviceName))
     {
-        IPropertyTree& service = itr->query();
-        //Only show the public services for now
-        if (!service.getPropBool("@public"))
-            continue;
+        VStringBuffer xpath("Software/RoxieCluster[@name=\"%s\"]", serviceName);
+        getRoxieService(environmentRoot->queryPropTree(xpath.str()), services);
```

Comment:
serviceName is coming from a user query. Supplying an invalid value will crash the called function (since queryPropTree() will return null).

Reply:
Will add a check.